app/src/main/java/app/revanced/integrations/patches/OldQualityLayoutPatch.java

```diff
@@ -1,52 +1,21 @@
 package app.revanced.integrations.patches;
 
-import android.graphics.Color;
-import android.graphics.drawable.ColorDrawable;
-import android.util.Log;
-import android.view.View;
-import android.view.ViewGroup;
-import android.view.Window;
+import android.os.Handler;
+import android.os.Looper;
 import android.widget.ListView;
-import android.widget.RelativeLayout;
-
-import java.util.Arrays;
+import android.view.View;
 
 import app.revanced.integrations.settings.SettingsEnum;
-import app.revanced.integrations.utils.LogHelper;
 
 public class OldQualityLayoutPatch {
-    public static Window window;
-    private static boolean hideWindow = true;
-    
     public static void showOldQualityMenu(ListView listView)
     {
-        if (!SettingsEnum.OLD_STYLE_QUALITY_SETTINGS.getBoolean()) return;
-
-        hideWindow = true;
-
-        listView.setOnHierarchyChangeListener(new ViewGroup.OnHierarchyChangeListener() {
-            @Override
-            public void onChildViewAdded(View parent, View child) {
-                LogHelper.debug(OldQualityLayoutPatch.class, "Added: " + child);
+        listView.setVisibility(View.GONE);
 
-                parent.setVisibility(View.GONE);
-                if (window != null && hideWindow) {
-                    window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
-
-                    hideWindow = false;
+        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
+                @Override public void run() {
+                        listView.performItemClick(null, 4, 0);
```

Comment:
This is not an elegant solution. It is error-prone as sleeping 1 millisecond might be too fast for example. Instead, when the child view is added, this event should be triggered.

Reply:
I agree on that. I'm going to close the PR, @OxrxL please reopen when you added those changes

Reply:
I already explained why this solution is safe and glitch-free. respect the current one.

There's no other way to do that and obtain the same result.

Reply:
> I agree on that. I'm going to close the PR, @OxrxL please reopen when you added those changes

There's no other way to get a satisfying solution, without this approach.

However, you are free to keep something that works badly.